### PR TITLE
[13.0][IMP] mrp_unbuild_bom_cust_qty: Change product_qty function to be reu…

### PR DESCRIPTION
**Pull request by adriresu**
…sable

### CODE REVIEW

File: mrp_unbuild_bom_cust_qty/models/mrp_unbuild.py [hunk 1]
```diff
@@ -222,12 +222,15 @@ def _update_product_qty_from_bom_totals(self):
...
             return
         for unbuild in self:
-            total_product_qty = sum(
-                item.product_uom_id._compute_quantity(
-                    item.total_qty, unbuild.product_uom_id
-                )
-                for item in unbuild.bom_quants_total_ids.filtered(
-                    lambda x: x.product_uom_id.category_id == unbuild.product_uom_id.category_id
-                )
+            unbuild.product_qty = self._get_product_qty_from_bom_totals() if self._get_product_qty_from_bom_totals() > 0 else 1
```
Comment (dalonsod) [inline mrp_unbuild_bom_cust_qty/models/mrp_unbuild.py]: With this line total calculation is unnecessary made twice. Please reduce processing:
```suggestion
            total_qty = self._get_product_qty_from_bom_totals()
            unbuild.product_qty = total_qty if total_qty > 0 else 1
```

**Review by dalonsod — CHANGES_REQUESTED** [review]

File: mrp_unbuild_bom_cust_qty/models/mrp_unbuild.py [hunk 2]
```diff
@@ -222,12 +222,16 @@ def _update_product_qty_from_bom_totals(self):
...
             return
         for unbuild in self:
-            total_product_qty = sum(
-                item.product_uom_id._compute_quantity(
-                    item.total_qty, unbuild.product_uom_id
-                )
-                for item in unbuild.bom_quants_total_ids.filtered(
-                    lambda x: x.product_uom_id.category_id == unbuild.product_uom_id.category_id
-                )
+            total_qty = self._get_product_qty_from_bom_totals()
```
Comment (dalonsod) [inline mrp_unbuild_bom_cust_qty/models/mrp_unbuild.py]: Grrr don't use `self` within loop and `multi` method!!

Comment (adriresu) [reply mrp_unbuild_bom_cust_qty/models/mrp_unbuild.py]: Done

**Comment by dalonsod** [comment]
OK, could you squash commits?

@ChristianSantamaria could we test it in AluTest environment?

**Comment by ChristianSantamaria** [comment]
@dalonsod Squashed, rebased and tested in Test enviroment :+1:

**Comment by dalonsod** [comment]
Great! Added @adriresu to contributors and extra commit with README update

**Pull request merged**